core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -343,11 +343,15 @@ public void commit() {
         LOG.warn("Failed to load committed snapshot, skipping manifest clean-up");
       }
 
-    } catch (RuntimeException e) {
-      LOG.warn("Failed to load committed table metadata, skipping manifest clean-up", e);
+    } catch (Throwable e) {
+      LOG.warn("Failed to load committed table metadata or during cleanup, skipping further cleanup", e);
```

Comment:
Failed loading committed table metadata or during cleanup

or

Failed to load committed table metadata or failed during cleanup